src/main/java/org/broadinstitute/hellbender/utils/mcmc/ParameterizedState.java

```diff
@@ -1,36 +1,104 @@
 package org.broadinstitute.hellbender.utils.mcmc;
 
-import java.util.List;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.*;
 
 /**
- * Basic implementation of {@link AbstractParameterizedState} with no convenience fields or getters.  See
- * example of use in GibbsSamplerSingleGaussianUnitTest.
+ * Represents a mapped collection of {@link Parameter} objects, i.e., named, ordered, enumerated keys associated with
+ * values of mixed type via a key -> key, value map.
+ * See GibbsSamplerSingleGaussianUnitTest and GibbsSamplerCopyRatioUnitTest for examples of use.
  *
  * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
  */
-public final class ParameterizedState extends AbstractParameterizedState {
+public class ParameterizedState<T extends Enum<T> & ParameterEnum> {
+    private final LinkedHashMap<T, Parameter<T, ?>> parameterMap;     //must be a Map that gives a sorted keySet
+
     /**
-     * Subclasses of {@link AbstractParameterizedState} must implement the
-     * {@link AbstractParameterizedState#copy(Class)} method.  This boilerplate implementation simply calls the
-     * copy constructor {@link ParameterizedState#ParameterizedState(ParameterizedState)}.
+     * Constructs a {@link ParameterizedState} with parameters enumerated by {@link ParameterEnum}
+     * from a List of {@link Parameter} objects with values of mixed type,
+     * checking that all parameters are present without duplicates.
+     * @param parameters    List of Parameters
+     * @throws IllegalArgumentException if {@code parameters} is null, empty, contains duplicate parameter names, or missing parameter names
      */
-    @Override
-    protected <S extends AbstractParameterizedState> S copy(final Class<S> stateClass) {
-        return stateClass.cast(new ParameterizedState(this));
+    public ParameterizedState(final List<Parameter<T, ?>> parameters) {
+        Utils.nonNull(parameters, "List of parameters cannot be null.");
+        Utils.nonEmpty(parameters, "List of parameters cannot be empty.");
+        final LinkedHashMap<T, Parameter<T, ?>> map = new LinkedHashMap<>();
```

Comment:
Since this constructor is not performance-sensitive, I would separate the logic of 1) checking for duplicates and 2) adding parameters.  Something like
```java
final int numUniqueParameters = parameters.stream().map(Parameter::getName).distinct().count();
Utils.validateArg(numUniqueParameters == parameters.size(), "Parameters may not contain duplicates.");
final LinkedHashMap<T, Parameter<T, ?>> map = new LinkedHashMap<>();
parameters.stream().forEach(p -> map.put(p.getName(), p);
```